### PR TITLE
CI binary release by tag

### PR DESCRIPTION
**Pull request by finiteops**

### CODE REVIEW

File: src/register-evm-agent/src/register_evm_agent/Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ hex = { workspace = true }
 ic-agent = { workspace = true }
 ic-exports = { workspace = true, features = ["state-machine"] }
 log = { workspace = true }
+openssl = {version="0.10.55", features = ["vendored"] }
```
Comment (veeso) [inline src/register-evm-agent/src/register_evm_agent/Cargo.toml:34]: why is this necessary?

**Review by ufoscout — APPROVED** [review]

**Review by itsyaasir — APPROVED** [review]

**Review by veeso — APPROVED** [review]

File: src/register-evm-agent/src/register_evm_agent/Cargo.toml (hunk 1 above)
Comment (finiteops) [reply src/register-evm-agent/src/register_evm_agent/Cargo.toml:34]: Openssl requires native libs. Thus there is a big possibility that linux build would work only for the platform only compatible with CI build OS. On different linux distros actual openssl libs could be located by different paths.

Comment (ufoscout) [reply src/register-evm-agent/src/register_evm_agent/Cargo.toml:34]: We should verify what dependency requires openssl and check if there's a feature to use rustls instead

Comment (finiteops) [reply src/register-evm-agent/src/register_evm_agent/Cargo.toml:34]: Cargo tree shows openssl in dependencies. 

BTW this one reason why I just can't easily cross compile it. I need to cross compile  openssl for all target platforms too.

**Review by veeso — APPROVED** [review]
